### PR TITLE
Property accessor is not recursive when called on another object

### PR DESCRIPTION
**Pull request by nd**

### CODE REVIEW

File: idea/src/org/jetbrains/kotlin/idea/inspections/RecursivePropertyAccessorInspection.kt [hunk 1]
```diff
@@ -96,6 +97,9 @@ class RecursivePropertyAccessorInspection : AbstractKotlinInspection() {
             val bindingContext = element.analyze()
             val target = bindingContext[REFERENCE_TARGET, element]
```
Comment (stephanienobles) [inline idea/src/org/jetbrains/kotlin/idea/inspections/RecursivePropertyAccessorInspection.kt]: y :-*

File: idea/src/org/jetbrains/kotlin/idea/inspections/RecursivePropertyAccessorInspection.kt [hunk 2]
```diff
@@ -96,6 +97,9 @@ class RecursivePropertyAccessorInspection : AbstractKotlinInspection() {
...
             if (target != bindingContext[BindingContext.DECLARATION_TO_DESCRIPTOR, propertyAccessor.property]) return false
+            (element.parent as? KtDotQualifiedExpression)?.let {
+                if (it.receiverExpression.text != KtTokens.THIS_KEYWORD.value && !it.isCompanionObjectReceiver(bindingContext)) return false
+            }
             return isSameAccessor(element, propertyAccessor.isGetter)
```
Comment (stephanienobles) [inline idea/src/org/jetbrains/kotlin/idea/inspections/RecursivePropertyAccessorInspection.kt]: sufjfubi

Comment (ivanljt) [inline idea/src/org/jetbrains/kotlin/idea/inspections/RecursivePropertyAccessorInspection.kt]: - 1. 1. ~~~~~~~~ ~~ ~~ 
~~__
1. ### _______ _ _ _ _ _ __ ~~ *

_Sent from my Meizu m1 metal using [FastHub](https://play.google.com/store/apps/details?id=com.fastaccess.github)_

Comment (ivanljt) [inline idea/src/org/jetbrains/kotlin/idea/inspections/RecursivePropertyAccessorInspection.kt]: @stephanienobles 

_Sent from my Meizu m1 metal using [FastHub](https://play.google.com/store/apps/details?id=com.fastaccess.github)_

**Comment by mglukhikh** [comment]
Slightly changed and merged into master. Thank you.

**Pull request closed without merging**